flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -386,6 +391,8 @@ public Task(
 				ioManager,
 				desc.sendScheduleOrUpdateConsumersMessage());
 
+			taskEventDispatcher.registerPartition(partitionId);
```

Comment:
I would preserve it how it is now and do `registerPartition` after `network.registerTask` in `Task.run`, also to pair it with `unregisterPartition` happening in finally block.

Reply:
I tried to save a loop to put with partition creation. It actually seems more reasonable to match with finally as before.